### PR TITLE
Display friendly 404 pages

### PR DESCRIPTION
**Pull request by gjvis**
Displays a friendly 404 page. Content validated by Vanessa.

Breadcrumbs include a link to the new search page, either within the current scrapbook or a new one, depending on the current path.

The "x saved" link is displayed if possible.

### CODE REVIEW

File: features/custom_error_pages.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@allow-rescue
+@allow-error-pages
```
Comment (gjvis) [inline features/custom_error_pages.feature]: By default Cucumber surfaces exceptions to ease debugging. These tags trigger hooks that allow production-like exception handling behaviour (i.e. friendly error pages).

File: config/application.rb [hunk 2]
```diff
@@ -32,5 +32,6 @@ class Application < Rails::Application
     config.active_record.raise_in_transactional_callbacks = true
     config.action_view.embed_authenticity_token_in_remote_forms = true
     config.autoload_paths << Rails.root.join("lib")
+    config.exceptions_app = self.routes
```
Comment (nathancashmoredwp) [inline config/application.rb]: What is self.routes

File: features/custom_error_pages.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+@vcr
+Feature: Custom error pages
+  In order to not get lost
```
Comment (nathancashmoredwp) [inline features/custom_error_pages.feature]: i like this...

**Comment by nathancashmoredwp** [comment]
Would be good to get a bit more of an overview when you have time.  thanks.
:bikini: :bikini: :bikini: :bikini: :bikini: :bikini: :bikini:

**Pull request merged**